digiroad2-api-oth/src/main/scala/fi/liikennevirasto/digiroad2/MunicipalityApi.scala

```diff
@@ -1,1132 +1,390 @@
+            }
+        case None =>
       }
-    }.get
-
-    getManoeuvreAndRoadLinks(manoeuvreIds)
-  }
-
-  def convertValidityPeriod(validityOpt: Option[ManoeuvreProperties]): Option[Set[ValidityPeriod]] = {
-    validityOpt match {
-      case Some(validityPeriod) => Some(validityPeriod.value.asInstanceOf[List[Map[String, Any]]].map { a =>
-        ValidityPeriod(a.find(a => a._1 == "startHour").map(_._2).get.asInstanceOf[BigInt].toInt,
-          a.find(a => a._1 == "endHour").map(_._2).get.asInstanceOf[BigInt].toInt,
-          ValidityPeriodDayOfWeek(a.find(a => a._1 == "days").map(_._2).get.asInstanceOf[String]),
-          a.find(a => a._1 == "startMinute").map(_._2).get.asInstanceOf[BigInt].toInt,
-          a.find(a => a._1 == "endMinute").map(_._2).get.asInstanceOf[BigInt].toInt)
-      }.toSet)
-      case _ => None
-    }
-  }
-
-  def createPointAssets(assets: JValue, typeId: Int): Seq[PersistedPointAsset] = {
-    val service = verifyPointServiceToUse(typeId)
-    val assets = typeId match {
-      case Obstacles.typeId =>
-        parsedBody.extractOpt[Seq[NewAssetValues]].getOrElse(Nil).map(x =>IncomingObstacleAsset( x.linkId, x.startMeasure.toLong, x.properties.map(_.value).head.toInt))
-      case PedestrianCrossings.typeId =>
-        parsedBody.extractOpt[Seq[NewAssetValues]].getOrElse(Nil).map(x =>IncomingPedestrianCrossingAsset( x.linkId, x.startMeasure.toLong))
-      case RailwayCrossings.typeId =>
-        parsedBody.extractOpt[Seq[NewAssetValues]].getOrElse(Nil).map(x =>
-          IncomingRailwayCrossingtAsset(
-            x.linkId,
-            x.startMeasure.toLong,
-              x.properties.find(_.name == "safetyEquipment").map { safetyEquipment => safetyEquipment.value.toInt }.get,
-              x.properties.find(_.name == "name").map { name => name.value },
-              x.properties.find(_.name == "railwayCrossingId").map { code => code.value }.get
-          ))
-      case TrafficLights.typeId =>
-        parsedBody.extractOpt[Seq[NewAssetValues]].getOrElse(Nil).map(x =>IncomingPedestrianCrossingAsset( x.linkId, x.startMeasure.toLong))
-    }
 
-    val links = roadLinkService.getRoadLinksAndComplementariesFromVVH(assets.map(_.linkId).toSet)
-    val insertedAssets = assets.foldLeft(Seq.empty[Long]){ (result, asset) =>
-      links.find(_.linkId == asset.linkId) match {
-        case Some(link) =>
-          service.toIncomingAsset(asset, link) match {
-            case Some(pointAsset) => result ++ Seq(service.create(pointAsset, user.username, link))
-            case _ => result
+      pavementClass match {
+        case Some(value) =>
+          val pavementClassValue = DynamicValue(DynamicAssetValue(Seq(DynamicProperty("paallysteluokka", "single_choice", required = false, Seq(DynamicPropertyValue(value.asInstanceOf[String].toInt))))))
+
+          val duplicate = pavedRoadService.getPersistedAssetsByLinkIds(PavedRoad.typeId, Seq(link.linkId))
+          duplicate.isEmpty match {
```

Comment:
`
if (duplicate.isEmpty) {
              val newPavementClassAsset = NewLinearAsset(link.linkId, 0, link.length, pavementClassValue, properties("sideCode").asInstanceOf[BigInt].intValue(), vvhClient.roadLinkData.createVVHTimeStamp(), None)
              pavedRoadService.create(Seq(newPavementClassAsset), PavedRoad.typeId, AwsUser)
} else
pavedRoadService.update(Seq(duplicate.head.id), pavementClassValue, AwsUser)
`

Reply:
Applied this suggestion